### PR TITLE
build(package): update referred packages to latest version

### PR DESCRIPTION
**Pull request by hamzamahmood**
The following PR updates the referred external packages in the core-lib and core-lib test projects to make sure that the C# core-lib is up-to-date and has the latest fixes and improvements.

Packages updated in **APIMatic.Core**:

- Microsoft.Json.Pointer
- Polly
- SonarAnalyzer.CSharp
- System.Collections.Immutable

Packages updated in **APIMatic.Core.Test**:

- Microsoft.NET.Test.Sdk
- NUnit3TestAdapter
- SonarAnalyzer.CSharp

### CODE REVIEW

File: APIMatic.Core/Utilities/TestHelper.cs [hunk 1]
```diff
@@ -503,16 +503,7 @@ private static bool DoesRightValContainsSameItems(JArray leftJArray, object righ
...
-                {
-                    containsJObject = true;
-                    break;
-                }
-            }
-            return containsJObject;
+            return jArray.Any(item => item is JObject);
         }
 
         private static bool IsArrayOfJObject(JArray jArray)
```
[the remark this answers is not in the document]
Comment (hamzamahmood) [reply APIMatic.Core/Utilities/TestHelper.cs:509]: Code refactored. No, the SonarAnalyzer didn't report any error/warning on this.

**Review by asadali214 — APPROVED** [review]
Approved!

**Review by asadali214 — APPROVED** [review]
Approved!

**Review by asadali214 — APPROVED** [review]
Approved

**Pull request merged**